src/bin/cybergrape/main.rs

```diff
@@ -56,30 +59,75 @@ fn main() -> Result<(), Box<dyn std::error::Error>> {
             )),
         ),
         Serial(serial_command) => (
-            serial_command.num_tags,
+            serial_command.num_tags as usize,
             // outfile is common to both commands, though it means slightly different things
             serial_command.outfile,
             // the serial command doesn't have any audio samples and doesn't need gain/range info
             None,
         ),
     };
 
-    let available_ports = SerialPort::available_ports().expect("Failed to get available ports");
-    let selected_port_opt = device_selector::run_device_selector(available_ports)?;
+    // Figure out what serial port our antena box is on
+    let available_ports = SerialPort::available_ports()?;
+    let selected_port_opt = gui::device_selector(available_ports)?;
     let selected_port = match selected_port_opt {
         Some(port) => port,
-        None => return Ok(()),
+        None => {
+            error!("Port selection failed");
+            return Ok(());
+        }
     };
 
     // Try to open the requested port and set its read timeout to infinity
     // (well, about 584,942,417,355 years, which is close enough)
-    let mut port = SerialPort::open(selected_port, 115200).expect("Failed to open port");
+    let mut port = SerialPort::open(selected_port, BAUD_RATE).expect("Failed to open port");
     port.set_read_timeout(std::time::Duration::MAX)
         .expect("Failed to set read timeout");
 
     let hdm = Arc::new(Mutex::new(Hdm::new()));
-    let mut accumulator = UpdateAccumulator::new(hdm.clone());
+    let th_hdm = hdm.clone();
+
+    listen_on_port(port, hdm);
+
+    if let Some((_sound_data, gains, ranges)) = audio_settings {
+        let sphericalizer = Sphericalizer::new(gains.into_iter().zip(ranges).collect());
+    } else {
+        let sphericalizer = Sphericalizer::new(vec![(1.0, 1.0); num_tags]);
+
+        let td_buf = TDBufMeta::new(num_tags);
+        let time_delta = Duration::from_secs(1).div_f64(update_rate as f64);
+
+        let accumulator = UpdateAccumulator::new(th_hdm);
+
+        let (buf, _) = gui::fold_until_stop((td_buf, accumulator), move |(mut buf, mut acc)| {
+            if let Some(update) = sphericalizer.query(&mut acc) {
```

Comment:
Wait for an update, and once we unblock and receive it, add it into the buffer and return the `(buf, acc)` pair to the folder!